src/ProductConstructionService/ProductConstructionService.WorkItems/WorkItemConfiguration.cs

```diff
@@ -57,6 +57,7 @@ public static void AddWorkItemProcessor<TWorkItem, TProcessor>(this IServiceColl
         where TProcessor : IWorkItemProcessor<TWorkItem>
     {
         services.TryAddTransient(typeof(TProcessor));
+        services.TryAddKeyedTransient<IWorkItemProcessor>(typeof(TWorkItem));
```

Comment:
this would have to be modified tho, to register a TProcessor, it doesn't do anythign now